src/uct/rocm/Makefile.am

```diff
@@ -9,13 +9,14 @@ SUBDIRS = . gdr
 
 module_LTLIBRARIES      = libuct_rocm.la
 libuct_rocm_la_CPPFLAGS = $(BASE_CPPFLAGS) $(ROCM_CPPFLAGS)
-libuct_rocm_la_CFLAGS   = $(BASE_CFLAGS)
+libuct_rocm_la_CFLAGS   = $(BASE_CFLAGS) -march=native
```

Comment:
This should come from top level "optimized" configure option. If you have to specify this here explicitly - something is broken.

Reply:
@shamisp Agreed. I didn't realize that the "release" configure option does not enable optimizations.

Reply:
@souravzzz - release option is used by distros. Since they target broad distribution, we do not enable u-arch optimizations in the release build, since it is not portable. We do enable bunch of other optimizations which are not u-arch specific.

Reply:
@shamisp Does UCX explicitly disallow the scenario where a feature like SSE/AVX is available at compile time (built on a newer system) but not at run time (run on an older system)? src/ucs/sys/init.c:23 `ucs_check_cpu_flags()` seems to throw an error if this happens. If so, we don't need the runtime checks.